### PR TITLE
Io tests

### LINKED ISSUE #71: [Feature Request] clean up io tests and plan a new set
### **Is your feature request related to a problem? Please describe.**

Currently IO tests were created ad-hoc and not really planned. These should get cleaned up and planned out in a senseful way.

### **Describe the solution you'd like**

We must cover:
- Reading/Writing to/from different io libraries
- Tables

So for FsSpreadsheet.ExcelIO we must test:
- [x] READ fom exceljs
- [x] READ from excel (the app)
- [x] READ from ClosedXML
- [x] READ from libre
- [x] READ from FsSpreadsheet
- [x] ROUNDABOUT FsSpreadsheet.ExcelIO
- [x] WRITE 

ForFsSpreadsheet.Exceljs we must test:
- [x] READ fom FsSpreadsheet.ExcelIO
- [x] READ from excel (the app)
- [x] ~~#77~~
- [x] READ from libre
- [x] READ from FsSpreadsheet
- [x] ROUNDABOUT FsSpreadsheet.Exceljs
- [x] WRITE 

### More

- All files must contain a table with different cell value types. We propose the following:

| Numbers | Strings     | DateTime | Boolean | ARCtrl Column           | ARCtrl Column                                              |
|---------|-------------|----------|---------|------------------------|------------------------------------------------------------|
| 1       | Hello       | any      | true | (A) This is part 1 of 2 | (A) This is part 2 of 2                                    |
| 2       | World       | any      | false |                        | Tests if column names with whitespace at end can be unique |
| 3       | Bye         | any      | true |                         |                                                            |
| 4       | Outer Space | any      | false |                        |                                                            |

- Duplicate this Table to TWO worksheets to also include multiple worksheets.

- Due to the nature of these tests, we can recycle most of the test files. We only need one file of any kind for read-in.

- All files written by tests should be prefixed with "WRITE_" so they can be deleted at any point, while all other files MUST never be opened AND saved to avoid corrupting their origin.

### PR DESCRIPTION
**Pull request by Freymaurer**
closes #71 

- I changed FsCell.Value to be obj. This should pave the way for future support of for example hyperlink/format cell values. In addition this change should reduce ambiguity. For example: boolean could be stored as any of `"true"`, `"TRUE()"`, `"1"`,... not it will be stored as true boolean value `true`.

### CODE REVIEW

File: src/FsSpreadsheet.ExcelIO/Cell.fs [hunk 1]
```diff
@@ -137,26 +137,21 @@ module Cell =
...
             |> fun c ->
                 if value.EndsWith " " then
                     setSpacePreserveAttribute c
                 else c
 
-    /// <summary>
-    /// Create a cell using a shared string table, also returns the updated shared string table.
-    /// </summary>
-    let fromValueWithDataType (sharedStringTable : SharedStringTable Option) columnIndex rowIndex (value : string) (dataType : DataType) = 
+    let getCellContent (doc : Packaging.SpreadsheetDocument) (value : string) (dataType : DataType) =
```
Comment (HLWeil) [inline src/FsSpreadsheet.ExcelIO/Cell.fs:146]: I don't like this name anymore.. 
Any idea for a better one?

File: src/FsSpreadsheet.Exceljs/Cell.fs [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+            fsCell.Value |> box |> Some 
+
+    /// <summary>
+    /// `worksheetName`, `rowIndex` and `columnIndex` are only used for debugging.
+    /// </summary>
+    /// <param name="worksheetName"></param>
+    /// <param name="rowIndex"></param>
+    /// <param name="columnIndex"></param>
+    /// <param name="jsCell"></param>
+    let readToFsCell worksheetName rowIndex columnIndex (jsCell: Cell) =
```
Comment (HLWeil) [inline src/FsSpreadsheet.Exceljs/Cell.fs:44]: Naming (I think this is better as the one in ExcelIO). 

Maybe just let as is for now and clean up when the actual need arises.

**Review by HLWeil — APPROVED** [review]
Now I can sleep tight again. I think it is already good as is with all these wonderful tests. 

I would postpone the code readability improvements. As this PR is already bulky enough.

File: build/TestTasks.fs
[the remark this answers is not in the document]
Comment (Freymaurer) [reply build/TestTasks.fs:1]: Yeah this is meant to improve automatisation, but has issues, as cmd files executed from fake cannot find the files they should run themselves. I think at the moment, it is not required to add this for test automatisation.

File: src/FsSpreadsheet.ExcelIO/Cell.fs (hunk 1 above)
Comment (Freymaurer) [reply src/FsSpreadsheet.ExcelIO/Cell.fs:146]: Which function name do you specifically mean by this?

Comment (HLWeil) [reply src/FsSpreadsheet.ExcelIO/Cell.fs:146]: `getCellContent`
It's not really clear whether it's about getting the content of an FsCell or an OpenDocument Cell.

**Comment by HLWeil** [comment]
BTW: All test files can be opened without warnings in Excel

**Pull request merged**